src/helpers/isEqual.js

```diff
@@ -0,0 +1,33 @@
+const isEqual = (first, second) => {
+  /* Checking if the types and values of the two arguments are the same. */
+  if (first === second) return true;
+
+  /* Checking if any arguments are null */
+  if (first === null || second === null) return false;
```

Comment:
What's about `undefined`? Why does that not get a similar special handling?

Reply:
> What's about `undefined`? Why does that not get a similar special handling?

Actually I am thinking about removing  this line. Both object could very well be null or undefined and if the are, they are equal.  Does that make sense?